### ISSUE TITLE
create_node() corrupts the DOM of an existing node that previously used the ID of the new node

### ISSUE THREAD

**Issue by AdamReece-WebBox** [vakata/jstree#1830]
I'm developing a dynamically editable data structure using JsTree to visualise it and to choose a data node to edit.

I noticed that after calling `create_node()` to create a 2nd node using an auto-generated ID it can corrupt older nodes in the same branch if those old nodes previously used the new node ID. This means the new node can't be selected because the anchors of the old node are wrong.

Example situation with DOM outputs:

Add a new node, it has an auto-generated ID of "new-field-1"...

```
<ul class="jstree-children" role="group">
    <li aria-labelledby="new-field-1_anchor" aria-level="2" aria-selected="false" class="jstree-node jstree-leaf jstree-last" id="new-field-1" role="treeitem">
        <i class="jstree-icon jstree-ocl" role="presentation"></i>
        <a class="jstree-anchor" href="#" id="new-field-1_anchor" tabindex="-1">
            <i class="jstree-icon jstree-themeicon fa fa-folder-open jstree-themeicon-custom" role="presentation"></i>
            new-field-1
        </a>
    </li>
</ul>
```

Now change the ID using `set_id()` of this field to anything, such as "address"...

```
<ul class="jstree-children" role="group">
    <li aria-labelledby="address_anchor" aria-level="2" aria-selected="false" class="jstree-node jstree-leaf jstree-last" id="address" role="treeitem">
        <i class="jstree-icon jstree-ocl" role="presentation"></i>
        <a class="jstree-anchor" href="#" id="address_anchor" tabindex="-1">
            <i class="jstree-icon jstree-themeicon fa fa-folder-open jstree-themeicon-custom" role="presentation"></i>
            address
        </a>
    </li>
</ul>
```

This is good. The old ID, "new-node-1", is nowhere to be seen.

Add a new node, it has an auto-generated ID of "new-field-1" because we had changed the ID the previous node using this ID.

```
<ul class="jstree-children" role="group">
    <li aria-labelledby="new-field-1_anchor" aria-level="2" aria-selected="false" class="jstree-node jstree-leaf" id="address" role="treeitem">
        <i class="jstree-icon jstree-ocl" role="presentation"></i>
        <a class="jstree-anchor" href="#" id="new-field-1_anchor" tabindex="-1">
            <i class="jstree-icon jstree-themeicon fa fa-folder-open jstree-themeicon-custom" role="presentation"></i>
            address
        </a>
    </li>
    <li aria-labelledby="new-field-1_anchor" aria-level="2" aria-selected="false" class="jstree-node jstree-leaf jstree-last" id="new-field-1" role="treeitem">
        <i class="jstree-icon jstree-ocl" role="presentation"></i>
        <a class="jstree-anchor" href="#" id="new-field-1_anchor" tabindex="-1">
            <i class="jstree-icon jstree-themeicon fa fa-folder-open jstree-themeicon-custom" role="presentation"></i>
            new-field-1
        </a>
    </li>
</ul>
```

Somehow the first node we had changed the ID of to "address" has somehow gotten lots of "new-field-1" bits back. This makes the 2nd node, that should really have ID "new-node-1", un-selectable because bits of the 1st node also have this same ID.

This happens when `create_node()` for the 2nd node finishes.

Tried on versions 3.2.1 and 3.3.4.

**Comment by vakata** [comment]
Here is a demo: http://jsfiddle.net/2kwkh2uL/6420/
As you can see the anchor ID and the aria-labelledby are never changed. And each new node is generated sequentially - there is no way to generate the same ID (if relying on jstree's ID generation).

What you are actually doing is generating and reusing your own IDs (reusing ones that you have changed), like so:
http://jsfiddle.net/2kwkh2uL/6421/

While I recognize this as an issue, I will not be fixing it for now - it is way too limited and easy to avoid. I am spending what little time I have on preparing v.4 for release. I will get round to this eventually but in the mean time - either make sure your IDs are not reused, or rely on jstree's IDs.

**Comment by AdamReece-WebBox** [comment]
Thanks for your feedback. I've found a workaround for this based on what you've said.

As I know what the new node's ID will be in advance I call `create_node()` first without specifying an `id` property in the node object, and then call `set_id()` to give it the intended ID after using the response from `create_node()`.

This appears to work OK. -- The only difference is that I found the sorting wasn't called immediately after, but it did get called a couple of clicks around to do other things in the tree.
